File: gaia-sdk-python/gaia_sdk/graphql/request/type/KeyOne.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry


class KeyOne(list):
    """
    This entity represents the output of a delete impulse
    """

    def identity_id(self):
        self.append(lambda x: "identityId")

    def reference(self):
        self.append(lambda x: "reference")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))

File: gaia-sdk-javascript/src/graphql/request/type/KeyOne.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * This entity represents the output of a delete impulse
 */
export class KeyOne extends Array<(_:VariableRegistry) => string> {
public _typeName = "KeyOne";
    public identityId = () => { 
        this.push(_ => "identityId")
    };

    public reference = () => { 
        this.push(_ => "reference")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

